src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -256,14 +279,58 @@ public static boolean onLivingUseTotem(LivingEntity entity, DamageSource damageS
         return !NeoForge.EVENT_BUS.post(new LivingUseTotemEvent(entity, damageSource, totem, hand)).isCanceled();
     }
 
-    public static float onLivingHurt(LivingEntity entity, DamageSource src, float amount) {
-        LivingHurtEvent event = new LivingHurtEvent(entity, src, amount);
-        return (NeoForge.EVENT_BUS.post(event).isCanceled() ? 0 : event.getAmount());
+    /**
+     * Creates and posts an {@link LivingDamageEvent.Pre}. This is invoked in
+     * {@link LivingEntity#actuallyHurt(DamageSource, float)} and {@link Player#actuallyHurt(DamageSource, float)}
+     * and requires access to the internal field {@link LivingEntity#damageContainers} as a parameter.
+     *
+     * @param entity    the entity to receive damage
+     * @param container the container object holding the final values of the damage pipeline while they are
+     *                  still mutable
+     * @return the current damage value to be applied to the entity's health
+     *
```

Comment:
extra whitespace line (I think spotless should catch this, but I guess it doesn't)

Reply:
that's not enough words to new-line.  i'll inline it.